servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -436,7 +440,7 @@ private BlockMetadataManagerView getUpdatedView() {
   /** This method must be guarded by WRITE lock of mEvictionLock */
   private void freeSpaceInternal(long userId, long availableBytes, BlockStoreLocation location)
       throws IOException {
-    EvictionPlan plan = mEvictor.freeSpace(availableBytes, location);
+    EvictionPlan plan = mEvictor.freeSpaceWithView(availableBytes, location, getUpdatedView());
```

Comment:
I think you need to synchronize on the pin list lock in the `getUpdatedView` method.